.github/workflows/check.yml

```diff
@@ -13,34 +13,91 @@ jobs:
+    - name: Build with testing
+      run: ./gradlew --stacktrace :embulk-output-mysql:check
+      env:
+        _JAVA_OPTIONS: "-Xmx2048m -Xms512m"
+        EMBULK_OUTPUT_MYSQL_TEST_CONFIG: "${{ github.workspace }}/ci/mysql.yml"
+    - uses: actions/upload-artifact@v4
+      if: always()
+      with:
+        name: mysql5_7
+        path: embulk-output-mysql/build/reports/tests/test
+  mysql8_3:
+    runs-on: ubuntu-latest
+    # push: always run.
+    # pull_request: run only when the PR is submitted from a forked repository, not within this repository.
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
+    strategy:
+      fail-fast: false
+    services:
+      mysql:
+        # Testing with MySQL 8.3 here, neither with 8.4 nor 9.0, because
+        # MySQL 8.4 has disabled mysql_native_password by default.
+        # Connector/J 5.x cannot connect to MySQL 8.4 due to this.
+        # TODO: Start testing with MySQL 8.4 and/or 9.0 with Connector/J 8.x.
+        image: mysql:8.3
+        options: --health-cmd "mysqladmin ping -h localhost" --health-interval 20s --health-timeout 10s --health-retries 10
+        ports:
+        - "3306:3306"
         env:
-          _JAVA_OPTIONS: "-Xmx2048m -Xms512m"
-          EMBULK_OUTPUT_MYSQL_TEST_CONFIG: "${{ github.workspace }}/ci/mysql.yml"
-      - uses: actions/upload-artifact@v3
-        if: always()
-        with:
-          name: mysql
-          path: embulk-output-mysql/build/reports/tests/test
-  postgresql:
+          MYSQL_ROOT_PASSWORD: root
+          MYSQL_USER: ci
+          MYSQL_PASSWORD: password
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up JDK 8
+      uses: actions/setup-java@v4
+      with:
+        java-version: 8
+        distribution: "zulu"
+        cache: "gradle"
+    - name: Connect
+      run: mysql -h 127.0.0.1 --port 3306 -uroot -proot -e "show databases;"
+    - name: show version
+      run: mysql -h 127.0.0.1 --port 3306 -uroot -proot -e "select version();"
+    - name: Create database
+      run: mysql -h 127.0.0.1 --port 3306 -uroot -proot -e "create database ci_test;"
+    # Workaround to change MySQL's password mechanism to `mysql_native_password`
+    # from `caching_sha2_password` that is the default in MySQL 8 because
+    # Connector/J 5.x does not support `caching_sha2_password`.
+    # TODO: Start testing with `caching_sha2_password` with Connector/J 8.x.
+    #
+    # See also: https://dev.mysql.com/doc/refman/8.4/en/account-names.html
+    - name: Change password (root@localhost)
+      run: mysql -h 127.0.0.1 --port 3306 -uroot -proot -e "ALTER USER 'root'@'localhost' IDENTIFIED WITH mysql_native_password BY 'root';"
+    - name: Change password (root@%)
+      run: mysql -h 127.0.0.1 --port 3306 -uroot -proot -e "ALTER USER 'root'@'%' IDENTIFIED WITH mysql_native_password BY 'root';"
+    - name: FLUSH PRIVILEGES
```

Comment:
Is this `FLUSH PRIVILEGES` required to make the password (mechanism) changes effective? If then, can you explain it in the comment?

Reply:
I've misunderstood about flush privileges. It was not require. So, I remove it.